design/pipeline-artifacts.md

```diff
@@ -83,6 +144,32 @@ Or to avoid downloading any of the artifacts at all:
 
 You can add files to any artifact multiple times in the same job, in different jobs in the same stage, and in different stages in the same pipeline, or you can `seal` an artifact at any time to prevent further files from being added.
 
+## Variables
+
+In addition to the new `$(Pipeline.Workspace)` variable, we introduce a variable per resource indicating where it was checked out on disk. For example:
```

Comment:
What about explicit download, when the artifact name is specified. There is no \"resource\" directory. Are variables needed?

Reply:
I think variables are potentially needed and easier in the case you download to a specific path and just want to easy way to locate something. It has been pretty standard practice for our product for a while and it is the way we provide for information about objects in the pipeline for tasks.  I am not opposed to leaving off the environment variables as we move forward in a lot of these cases as it just bloats the environment and may not generally be necessary.  However, I think we will need to have pipeline variables that can be pulled from the context using either expressions or macros that cover the metadata about each of the resources.

Reply:
Eric, was yours a technical concern or customer/usability concern? I agree with Chris this is a good thing for the customer to have, but if it's complicated to build then might not be worth it.